pkg/storage/stores/shipper/compactor/compactor_test.go

```diff
@@ -94,7 +94,9 @@ func TestCompactor_RunCompaction(t *testing.T) {
 		testutil.SetupDBTablesAtPath(t, name, tablesCopyPath, dbs, false)
 	}
 
-	compactor := setupTestCompactor(t, tempDir)
+	cm := storage.NewClientMetrics()
+	// defer am.Unregister()
```

Comment:
why `defer` is commented? I think we can unregister() here? given we call this for each test?

Reply:
I would have expected this test to result in a duplicate metric registration panic but for some reason it isn't. Just to be safe I've fixed the unregister here.